File: frontend/src/types/barber.ts
export interface Barber {
  id: number;
  user_id: number;
  email: string;
  full_name: string;
  specialty: string;
  barbershop_name?: string | null;
  photo_url?: string | null;
  phone?: string | null;
  avatar?: string | null;
  rating?: number | null;
  average_rating?: number | null;
  years_experience?: number | null;
  completed_bookings_count?: number | null;
  price_from?: number | null;
  base_price?: number | null;
  latitude?: number | null;
  longitude?: number | null;
  address?: string | null;
  work_start_time?: string | null;
  work_end_time?: string | null;
  off_days?: string[];
  distance_km?: number | null;
  location_text?: string | null;
  services?: string[] | null;
  bio?: string | null;
  is_active: boolean;
  is_financially_blocked?: boolean;
  created_at: string;
  updated_at: string;
}

export interface AdminBarber extends Barber {
  total_bookings: number;
  today_bookings: number;
  balance?: number;
  debt?: number;
  commission_percent?: number;
}

export interface BarberFormPayload {
  full_name: string;
  specialty: string;
  barbershop_name?: string | null;
  photo_url: string;
  phone?: string | null;
  rating: number;
  years_experience: number;
  price_from?: number | null;
  base_price?: number | null;
  latitude?: number | null;
  longitude?: number | null;
  address?: string | null;
  work_start_time?: string | null;
  work_end_time?: string | null;
  off_days?: string[];
  email: string;
  password?: string;
  bio?: string | null;
  is_active?: boolean;
}

export interface AvailableSlot {
  time: string;
  is_available: boolean;
  is_booked: boolean;
  is_expired?: boolean;
}

export interface BarberService {
  id: number;
  barber_id: number;
  name: string;
  description?: string | null;
  price: number;
  duration_minutes: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface BarberScheduleItem {
  id?: number;
  weekday: number;
  start_time: string;
  end_time: string;
  break_start_time?: string | null;
  break_end_time?: string | null;
  is_working: boolean;
}

export interface BarberDayOff {
  id: number;
  barber_id: number;
  date: string;
  reason?: string | null;
}

export interface BarberVacation {
  id: number;
  barber_id: number;
  start_date: string;
  end_date: string;
  reason?: string | null;
}

File: backend/app/models/barber.py
from datetime import datetime, time

from sqlalchemy import Boolean, DateTime, Float, ForeignKey, Integer, JSON, String, Text, Time, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.user import Base


class Barber(Base):
    __tablename__ = "barbers"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), unique=True, nullable=False)
    full_name: Mapped[str] = mapped_column(String(140), nullable=False)
    specialty: Mapped[str] = mapped_column(String(180), nullable=False)
    barbershop_name: Mapped[str | None] = mapped_column(String(180))
    photo_url: Mapped[str] = mapped_column(String(500), nullable=False)
    phone: Mapped[str | None] = mapped_column(String(40))
    rating: Mapped[float] = mapped_column(Float, default=5.0, nullable=False)
    years_experience: Mapped[int] = mapped_column(Integer, default=1, nullable=False)
    price_from: Mapped[float | None] = mapped_column(Float)
    base_price: Mapped[float | None] = mapped_column(Float)
    latitude: Mapped[float | None] = mapped_column(Float)
    longitude: Mapped[float | None] = mapped_column(Float)
    address: Mapped[str | None] = mapped_column(String(255))
    work_start_time: Mapped[time | None] = mapped_column(Time)
    work_end_time: Mapped[time | None] = mapped_column(Time)
    off_days: Mapped[list[str] | None] = mapped_column(JSON)
    bio: Mapped[str | None] = mapped_column(Text)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    balance: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    debt: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    demo_balance: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    demo_debt: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    commission_percent: Mapped[int] = mapped_column(Integer, default=10, nullable=False)
    is_financially_blocked: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )

    user: Mapped["User"] = relationship(back_populates="barber")
    bookings: Mapped[list["Booking"]] = relationship(back_populates="barber", cascade="all, delete-orphan")
    working_hours: Mapped[list["WorkingHour"]] = relationship(back_populates="barber", cascade="all, delete-orphan")
    blocked_slots: Mapped[list["BlockedSlot"]] = relationship(back_populates="barber", cascade="all, delete-orphan")
    services: Mapped[list["BarberService"]] = relationship(back_populates="barber", cascade="all, delete-orphan")
    day_offs: Mapped[list["BarberDayOff"]] = relationship(back_populates="barber", cascade="all, delete-orphan")
    vacations: Mapped[list["BarberVacation"]] = relationship(back_populates="barber", cascade="all, delete-orphan")
    favorited_by: Mapped[list["CustomerFavoriteBarber"]] = relationship(back_populates="barber", cascade="all, delete-orphan")
    transactions: Mapped[list["BarberTransaction"]] = relationship(back_populates="barber", cascade="all, delete-orphan")
    daily_settlements: Mapped[list["BarberDailySettlement"]] = relationship(
        back_populates="barber", cascade="all, delete-orphan"
    )
    demo_transactions: Mapped[list["DemoBarberTransaction"]] = relationship(
        back_populates="barber", cascade="all, delete-orphan"
    )
    demo_daily_settlements: Mapped[list["DemoDailySettlement"]] = relationship(
        back_populates="barber", cascade="all, delete-orphan"
    )
